x-pack/legacy/plugins/watcher/server/np_ready/models/action_status/action_status.js

```diff
@@ -58,20 +63,10 @@ export class ActionStatus {
       return ACTION_STATES.ERROR;
     }
 
-    // At this point, we cannot determine the action status so we thrown an error.
+    // At this point, we cannot determine the action status so mark it as "unknown".
     // We should never get to this point in the code. If we do, it means we are
     // missing an action status and the logic to determine it.
-    throw badImplementation(
-      i18n.translate(
-        'xpack.watcher.models.actionStatus.notDetermineActionStatusBadImplementationMessage',
-        {
-          defaultMessage: 'Could not determine action status; action = {actionStatusJson}',
-          values: {
-            actionStatusJson: JSON.stringify(actionStatusJson),
-          },
-        }
-      )
-    );
+    return ACTION_STATES.UNKNOWN;
```

Comment:
It seemed more appropriate to mark the action in an unknown state, rather than throwing an error and making the watch list view unusable. 

The only potential outstanding question is what the overall watch status should return in this scenario. I did not make any code changes to [how this is handled](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/watcher/server/np_ready/models/watch_status/watch_status.js#L51), so it currently will return `OK`.  Thoughts?

Reply:
I definitely agree with rather keeping the view usable in case of a bad status (i.e., catering for this condition here definitely seems like the correct thing to do). Overall I am happy with this approach :+1:

The one advantage of throwing in this case, however, is that we were able to discover this bug which I think may have remained hidden otherwise. So being totally silent also has a cost (as you've noted).

Judging how action_status and watch_status hang together, it would be cool to be able to say that if the _last_ action status was unknown, then the watch_status is unknown? Remaining in this state should be a red flag to the user + they can still use the UI, we should communicate it's a possible bug. This could be a separate unit of work to this fix though!

Let me know what you think? We could revert this line for now and address later, or commit and address later. I'm happy with either for now.

Reply:
@jloleysens I like this idea a lot - thanks! I opened https://github.com/elastic/kibana/issues/55410 to address separately.